cantusdb/turtle/csv2rdf_single_subject.py

```diff
@@ -0,0 +1,67 @@
+from rdflib import Graph, URIRef, Literal
+from rdflib.namespace import RDF
+import csv
+import validators
+import sys
+import json
+
+
+# a main operation function
+def convert_csv_to_turtle(filename) -> Graph:
+    """
+    Adds all informations as RDF triples from the input filename into a graph and return it.
+    @Pre: input filename must be type string.
+    @Post: Returns a RDF.Graph that has all the triples
+    """
+
+    if len(sys.argv) != 2:
+        raise ValueError("Invalid number of input filename")
+
+    with open(filename, "r", encoding="utf-8") as csv_file:
+        # TODO: maybe change this one to accomadate for multiple files
+
+        g = Graph()
+
+        csv_reader = csv.reader(csv_file)
+        with open("relations_mapping.json", "r") as mapper:
+            ontology_dict = json.load(mapper)
+
+        header = next(csv_reader)
+
+        # Convert each row to Turtle format and add it to the output
+        for row in csv_reader:
+            # the first column as the subject
+            key_attribute = URIRef(row[0])
+            if "type" in ontology_dict:
+                g.add((key_attribute, RDF.type, URIRef(ontology_dict["type"])))
+
+            # extracting other informations
+            # TODO: is the source description texts needed?
+            for i, element in enumerate(row[1:]):
+                predicate_cur = header[i + 1]
+                if element == "" :
+                    continue
+                
+                # finding the predicate from csv in the config dictionary, if not exit, skip
+                if predicate_cur not in ontology_dict:
+                    continue
+
+                predicate = URIRef(ontology_dict[predicate_cur])
+
+                # the object might be an URI or a literal
+                if validators.url(element):
+                    obj = URIRef(element)
+                else:
+                    obj = Literal(element)
+
+                g.add((key_attribute, predicate, obj))
+
+    return g
+
+
+# openrefine_csv_test.csv
```

Comment:
What is this comment saying? This file is not referenced in this script/function.

Reply:
This is the test file used for this script. This comment is here to ease copy and paste.

Reply:
But as another developer, who only has access to this repository and not any other files you only have on your local machine, I don't know what `openrefine_csv_test.csv` is. What do I need to run this script? A csv created through openrefine? If so, then I think that's what your doctring/argument help-text should say.